### PR TITLE
Manually start election

### LINKED ISSUE #58: Manually start an election
Ref: VOC-SEC-05

As an administrator, there are elections that don't have a defined start date time, but that start on an administrator action (for instance, when a face to face meeting starts). For these cases, I need to be able to manually start an election.

This means that we need to change the current form, so the start date configuration will have a checkbox, with three options:

1. Start on a specific date
2. Start manually

Each of these actions will work as this:

|Option|Description|
|---|---|
|1. Start on a specific date |Current behavior. When selected, a date time selector will appear that allows the admin to configure the `start_time` for the election. Regarding the `electionType` field, it'll set it up with an autoStart value of "true".|
|2. Start manually  |It will not start the election automatically. When selected, the date time selector will disappear. It will set up the election without any `start_time`. It'll set up the election with state "Paused" and it'll allow the administrator to "Start" the election with a button (similarly to the controls that we offer when the election is Interruptible, see #47). Regarding the `electionType` field, it'll set it up with an `autoStart` value of "false".

### Acceptance criteria

- [x] Given that I'm an administrator,
      When I create a new Election,
      Then I can create it with the "Start on a specific date" option,
      And I see the `start_time` date time picker when editing or creating the Election
      And when I set up the Election, this election starts at the configured date and time,
      And the start date is showed publicly.
- [x] Given that I'm an administrator,
      When I create a new Election,
      Then I can create it with the "Start manually" option,
      And I don't see the `start_time` date time picker when editing or creating the Election
      And when I set up the Election, this election doesn't start,
      And I have a button to "Start election",
      And the start date is showed publicly.
- [x] Given that I'm a visitor and there's an Election created with the "Start manually" option
      When I go to see a published election metadata (i.e. cards/lists and single election page)
      Then I see that this election doesn't have a start date
- [x] Given that I'm a developer,
      When I consume the GraphQL API,
      Then it works with each of these options
- [x] Given that I'm a developer,
      When I run the seeds,
      Then there are some elections with each of these options

### PR DESCRIPTION
**Pull request by antopalidi**
fixes #58, #66

In this PR it is implemented the feature of starting elections with the button

#### Manual start setting
- [x] The start time field is inactive when manual start is selected
- [x] When selecting manual start the field is set to values:
`auto_start: true`
`start_time: nil`
`interruptible: true`

Screenshot 2023-06-27 at 16 27 00

#### Setup Election
- [x] When an election with manual start is setting, the values are set:
`start_time: 30.seconds.from.now` – configurable
`election.status = :paused` – elections are paused automatically.
- [x] Add button for manual start
- [x] After a manual start, the election is set to `:vote` status

Screenshot 2023-06-27 at 16 25 14

- [x] Add manual_start to GraphQL API
- [x] Add options to seeds

### CODE REVIEW

**Review by andreslucena — CHANGES_REQUESTED** [review]
I have the first round of feedback, can you check it out?

**Comment by andreslucena** [comment]
@antopalidi if you agree with any of my suggestions please use the GitHub UI for accepting them (with the "Commit sugestion" or "Add suggestion to batch" buttons). This way will be easier and faster for you and the suggestions will be automatically marked as resolved conversations.

> One test fails on github, but as I understand it has no relation to my development.

They're related to #63, we probably missed them on that review. Can you fix them here though? So we have a green build before merging this one.

On another hand I think it's good that @carolromero also review this one so she can check the behavior. I'm not 100% if this should be marked by default and how it looks in the frontend before hitting the "Start election" button. Can you upload this branch to the staging server @antopalidi ?

**Comment by antopalidi** [comment]
@andreslucena 
I have a question, do we need the `:manual_start_time_setting` configuration if it is used during the setup and creation of elections on the blockchain to set the start time.

First, I was set the start time after configuring the calendar when the manual start option was selected. But I was getting an error during setup election: "cannot add process with start block lower than or equal to the current height"

Now I think maybe set the start time during setup election to `Time.zone.now` without using this config.

**Comment by andreslucena** [comment]
> "cannot add process with start block lower than or equal to the current height"

 The "blocks" in Vocdoni are how they actually keep track of time. I had this error when trying to set up an election with `start_time` in the past.

> Now I think maybe set the start time during setup election to Time.zone.now without using this config.

Try this out to see if it works multiple times. Mind that there's a delay between when we sent the operation and when they actually confirm that the operation was received and it's in the blockchain. You can check it out in the "Network" tab of your Web Development tools in your browser. So, I think that we may be bitten by this same error ("cannot add process with start block lower than or equal to the current height"). On another hand, we don't lose anything for trying it. 

For me, the 30 seconds delay is more than enough for waiting for the blockchain to actually confirm the operation.

**Comment by antopalidi** [comment]
> For me, the 30 seconds delay is more than enough for waiting for the blockchain to actually confirm the operation.

@andreslucena 
I tried it. There was no error.  But if it will save from a possible error, it is better to set 30 seconds. You are right.
Then I will add a description to the README. I changed the rest.

**Comment by carolromero** [comment]
@antopalidi thanks for all the work! 

here's my feedback after the review:

- [ ] In the pending steps shown on the dashboard, if you try to publish the election with "Fix it" before uploading the census, it shows you the message: "You are not authorized to perform this action":
image

This makes sense, but then we should change the order of the validations, and have them displayed like this (I have also changed a little bit the validation text of the manual election):
image


- [ ] I rewrote a little bit the message for the setup step:
image
  Here's the copy:

  Election successfully configured!
  At this point you can still make changes if you see something misconfigured. [Make sure that you preview the election.](https://decidim-vocdoni.dev.pokecode.net/assemblies/general-assembly/f/3/elections/9)

  If you are comfortable with the current configuration, click on the "Setup election" button to send it to the Vocdoni Blockchain. 
  Then the election will be able to start.
  Please note that this action is irreversible, with no further modifications allowed.

- [ ] Regarding the frontend, if an election configured to start manually, let's hide the message: Voting begins on "Manual start"
image

- [ ] Also in the frontend, I'm getting a server error when trying to view the election cards: https://decidim-vocdoni.dev.pokecode.net/assemblies/general-assembly/f/1/. I'd say that for now let's hide this section of the card if it is configured with manual start:
image

File: README.md [hunk 1]
```diff
@@ -60,9 +60,10 @@ By default, the module is configured to read the configuration from ENV variable
 Currently, the following ENV variables are supported:
 
 | ENV variable | Description | Default value |
-| ------------ | ----------- | ------------- |
+| ------------ | ----------- |-------|
 | VOCDONI_API_ENDPOINT_ENV | The environment of the Vocdoni API. Only two values are accepted: `dev`, `stg`. Read more on [Vocdoni SDK Usage Environment](https://github.com/vocdoni/vocdoni-sdk#environment) | `stg` |
-| VOCDONI_MINUTES_BEFORE_START | How many minutes should the setup be run before the election starts | `10` |
+| VOCDONI_MINUTES_BEFORE_START | How many minutes should the setup be run before the election starts | `10`  |
```
Comment (andreslucena) [inline README.md:65]: There's an extra space here that:
```suggestion
| VOCDONI_MINUTES_BEFORE_START | How many minutes should the setup be run before the election starts | `10` |
```

This kind of changes make difficult to work with git tooling (for instance git blame)

File: README.md [hunk 2]
```diff
@@ -60,9 +60,10 @@ By default, the module is configured to read the configuration from ENV variable
...
+| VOCDONI_MANUAL_START_TIME_SETTING_IN_SECONDS | How many seconds there will be a delay to start the election with manual start after creating it on the blockchain | `30`  |
```
Comment (andreslucena) [inline README.md:66]: There's an extra space here: 
```suggestion
| VOCDONI_MANUAL_START_TIME_SETTING_IN_SECONDS | How many seconds there will be a delay to start the election with manual start after creating it on the blockchain | `30` |
```

**Review by andreslucena — DISMISSED** [review]
LGTM! @microstudi do you want to give it an eye before merging?

**Review by microstudi — COMMENTED** [review]
I think we don't need the migration here as there's already an "auto_start" conf in the `election_type` field. We should change the backend to use only this field in order not to have repeated db data.

**Review by microstudi — APPROVED** [review]
All good now! Thanks everyone!

**Pull request merged**